lib/membrane_h264_plugin/parser.ex

```diff
@@ -231,10 +240,11 @@ defmodule Membrane.H264.Parser do
         {{[], state.au_splitter}, state.nalu_parser}
       end
 
-    {remaining_nalus, au_splitter} = AUSplitter.flush(au_splitter)
-    maybe_improper_aus = access_units ++ [remaining_nalus]
+    {last_au, au_splitter} = AUSplitter.flush(au_splitter)
```

Comment:
I thought in this case we assume that the AU we keep in the splitter state is finished and we _should_ flush it 🤔

Reply:
We flush it as there might be a complete NAL unit. That is also why we call the list with these flushed NALus attached a `maybe_improper_aus` :D 
Imagine a situation, in which the `:nalu_aligned` stream is "broken" in the following configuration `sps pps <break point> idr`.
The parser will receive `sps, pps, :end_of_stream` and on `handle_end_of_stream` the AUSplitter will flush `sps` and `pps` which do not form a complete AU.

Then, `prepare_actions_for_aus` will decide whether the passed list of NAL units is an access unit or not and in that case it will simply not return `sps` and `pps`.

Reply:
Agreed, but I think I didn't change that behaviour 🤔 is the comment about the name?

Reply:
indeed, it is :D However I have put some other comment concerning the part of updated code corresponding the part we are currently discussing, so I will resolve that comment